actionpack/lib/action_dispatch/railtie.rb

```diff
@@ -21,12 +21,17 @@ class Railtie < Rails::Railtie # :nodoc:
     config.action_dispatch.use_authenticated_cookie_encryption = false
```

Comment:
There's a part of me that thinks we should turn:

```ruby
config.action_dispatch.signed_cookie_salt = "signed cookie"
config.action_dispatch.encrypted_cookie_salt = "encrypted cookie"
config.action_dispatch.encrypted_signed_cookie_salt = "signed encrypted cookie"
config.action_dispatch.use_authenticated_cookie_encryption = false
```

into:

```ruby
config.action_dispatch.cookies = ActiveSupport::OrderedOptions.new
config.action_dispatch.cookies.signed_salt = "signed cookie"
config.action_dispatch.cookies.encrypted_salt = "encrypted cookie"
config.action_dispatch.cookies.encrypted_signed_salt = "signed encrypted cookie"
# These three are old 👆, but these three haven't shipped 👇.
config.action_dispatch.cookies.authenticated_encrypted_salt = "authenticated encrypted cookie"
config.action_dispatch.cookies.authenticated_encryption = false
config.action_dispatch.cookies.rotations = ActiveSupport::Messages::RotationConfiguration.new
```

We'd need to deprecate the old ones. Though I think it's highly unlikely that many apps have overriden those default salts, so the hurt isn't too huge. Either that or we just reassign for something so trivial.
It's probably out of scope for this, so let me look into that separately.

Though by the way, why do we need the `authenticated_encrypted_cookie_salt`? We could just as well reuse the `encrypted_signed_cookie_salt`?

Reply:
When I added AEAD support I wanted to use a different salt value as this is a best practice when deriving keys for different uses.